site/topics/genetic-algorithms/n-queens-ga.rst

```diff
@@ -14,12 +14,61 @@ Problem
 Initialization
 ==============
 
+* Before evolution can begin, an initial population of candidate solutions
+
+
 Representation
 --------------
 
+* The permutation representation previously discussed will be used
+
+    * A chromosome will be a list of length :math:`n`
+    * Each of the :math:`n` indices correspond to each of the :math:`n` queens
+    * Each index will be the queen's respective column in the chess board
+    * The elements within the list will be a permutation of the integers between :math:`0` and :math:`n-1`
+    * The value at each index will be the queen's respective row in the chess board
+
+
+* Since the values and indices are unique, no two queens can be in the same row or column
+
+* An example chromosome when :math:`n=8` is
+
+    ``<0, 5, 7, 6, 3, 2, 1, 4>``
+
+
 Population
 ----------
 
+* A population is a list of chromosomes
+
+    .. code-block:: text
+
+        [<0, 5, 7, 6, 3, 2, 1, 4>,
+         <4, 5, 2, 3, 6, 7, 1, 0>,
+         <0, 1, 2, 3, 4, 5, 6, 7>,
+         ...
+         ...
+         <4, 3, 2, 7, 6, 1, 0, 5>]
+
+
+
+* The chromosomes, and thus the population, will be randomly generated for this implementation of a genetic algorithm
+
+.. literalinclude:: /../src/ga_n_queens.py
+    :language: python
+    :lineno-match:
+    :start-after: # [begin-initialization]
+    :end-before: # [end-initialization]
+
+
+* The number of chromosomes within the population is defined by a hyperparameter called ``POPULATION_SIZE``
+* The length of the chromosomes will match the number of queens, which is defined by the hyperparameter ``N_QUEENS``
```

Comment:
Isn't a hyperparameter multidimensional? Cardinality and N are only scalar.

Reply:
The deep learning people started making "hyperparameter" mean a tunable parameter before the search starts as they use the word parameter for things like network weights that update during learning.

Reply:
...k. Not sure why they'd ever choose a word that already means higher dimensional, when things like _variable_ and _argument_ already exist.

Reply:
That's basically all of ML. They make up new words for things that already have a word. 

> At the beginning of this famous book "All of Statistics", the author has a table with the words the math/stats people use and the word the ML people use. It's amusing.

Reply:
![Screenshot_20230811-172326.png](https://github.com/jameshughes89/cs4XX-EvolutionaryComputation/assets/1328048/8bd772ad-50c6-4da1-ab49-8b63bcaf4d30)